### ISSUE TITLE
Lint "disallowed-methods" does not work with pointer methods.

### ISSUE THREAD

**Issue by Lorak-mmk** [rust-lang/rust-clippy#10156]
### Summary

I wanted to blacklist `as_ref()` / `as_mut()` methods on raw pointers, but it seems that it doesn't work with Clippy's "disallowed-methods".

### Reproducer

My config in clippy.toml:
```
disallowed-methods = [
    "std::ptr::as_ref",
    "std::ptr::as_mut",
    "core::ptr::const_ptr::as_ref",
    "core::ptr::mut_ptr::as_mut",
    "pointer::as_ref",
    "pointer::as_mut"
]
```

Example code:

```rust
fn main() {
    let x = 123;
    let ptr: *const i32 = &x;
    let x2 = unsafe { ptr.as_ref().unwrap() };
    println!("Hello, world! {x2}");
}

```

I expected Clippy to complain about `as_ref()` call but it didn't.

### Version

```text
rustc 1.66.0 (69f9c33d7 2022-12-12)
binary: rustc
commit-hash: 69f9c33d71c871fc16ac445211281c6e7a340943
commit-date: 2022-12-12
host: x86_64-unknown-linux-gnu
release: 1.66.0
LLVM version: 15.0.2
```

**Comment by Lorak-mmk** [comment]
Turns out you need to use `const_ptr::as_ref` and `mut_ptr::as_mut` - this should probably be documented somewhere.
